llama_index/embeddings/base.py

```diff
@@ -247,6 +253,16 @@ async def aget_queued_text_embeddings(
             embedding for embeddings in nested_embeddings for embedding in embeddings
         ]
 
+        for (event_id, text_batch), embeddings in zip(callback_payloads, nested_embeddings):
+            self.callback_manager.on_event_end(
+                CBEventType.EMBEDDING,
+                payload={
+                    EventPayload.CHUNKS: text_batch,
+                    EventPayload.EMBEDDINGS: embeddings,
+                },
+                event_id=event_id,
+            )
```

Comment:
mostly for my understanding - don't you want to call the `on_event_end` for all the chunks in one go here? Or is each embedding generation a separate event?

Reply:
Also question about the payload format - in this case the `text_batch` is actually a list correct?

Reply:
Also from a timing perspective, is the expectation not to call the callback as the embeddings are generated rather than all at once at the end?

Reply:
It could be either way, either logged here or in _aget_text_embeddings

Logging it in the above gives a bit more sense of "progress" observability I suppose 👀

Reply:
Yeah if timing of embedding generation is important, I think the progress will be important. Looks like timing information is the responsibility of the handler?

Reply:
Ah, I see now why we wait to the end, because otherwise the actual methods are still abstract. I think that's fine for now